core/src/com/cloud/storage/template/VhdProcessor.java

```diff
@@ -91,27 +97,58 @@ public long getVirtualSize(File file) throws IOException {
         return file.length();
     }
 
-    protected long getTemplateVirtualSize(File file) throws IOException {
+        protected long getTemplateVirtualSize(File file) throws IOException {
         byte[] currentSize = new byte[8];
+        byte[] cookie = new byte[8];
         byte[] creatorApp = new byte[4];
 
-        try (FileInputStream strm = new FileInputStream(file)) {
-            long skipped = strm.skip(file.length() - vhdFooterSize + vhdFooterCreatorAppOffset);
+
+        BufferedInputStream fileStream = new BufferedInputStream(new FileInputStream(file));
+        InputStream strm = fileStream;
+
+        boolean isCompressed = checkCompressed(file.getAbsolutePath());
+
+        if ( isCompressed ) {
+            try {
+                strm = new CompressorStreamFactory().createCompressorInputStream(fileStream);
+            } catch (CompressorException e) {
+                s_logger.info("error opening compressed VHD file " + file.getName());
+                return file.length();
+            }
+        }
+
+        try {
+
```

Comment:
Can you remove the unneeded line break here since I already have another change for you to make.  :)  Thx...